### DISCUSSION TITLE
Target rate sensor - Re-evaluate multiple times a day

### DISCUSSION THREAD

**Discussion by SteveGBuck** [BottlecapDave/HomeAssistant-OctopusEnergy#367] [Q&A]
Is there any detriment to enabling this option on a target rate sensor (for agile tariff?). My target rate sensor periods are from  16:00 - 16:00 and everything works well but I've noticed without re-evaluate option enabled the sensors lose their data if I restart home assistant until the next 16:00 schedule.

**Comment by BottlecapDave** [comment]
Hi there. If you enable that flag then there is a chance that the sensor will come on more than the period of time that you requested. This is because once the best time has passed, it will then find the next best time until the end time.

For example, if you were after a 1 hour slot between 4pm and 4pm and the best slot was between 3 and 4am. Once you hit 4:01am, it would then look for the next best slot between 4:01am and 4pm. The intention of this flag was two fold

1) In the early stages of the integration it was once the default behaviour, which was wrong
2) The above behaviour is useful when you have other sensors that determine if something is triggered, but it must run once a day. For instance the best time to run a vaccuum might be 8-9am one day, but there are other factors that mean its not the best time (e.g. you're in the house)

The fact the sensor does not restore after reload is a bug, and I'll create an issue from this to fix it.

**Reply by SteveGBuck** [reply]
Awesome. That makes sense and explains why one of my shorter sensors turned on at a time when the rate was relatively high!

And thanks. Yes, will look out for the bug fix!

**Reply by SteveGBuck** [reply]
Hey! Just wanted to note that I've applied your latest release today and initial signs look like this has worked!  Thank you!
